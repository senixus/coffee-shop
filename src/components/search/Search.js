import React from "react";
import "./search.scss";

const Search = ({ handleChange }) => {
  return (
    <div className="search">
      <div>
        <input
          type="text"
          placeholder="Search"
          name="coffee"
          onChange={handleChange}
          className="search-bar"
        />
      </div>
    </div>
  );
};

export default Search;
