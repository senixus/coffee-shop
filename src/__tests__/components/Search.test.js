import React from "react";
import { shallow } from "enzyme";
import Search from "../../components/search/Search";

it("expect to render Search component", () => {
  const props = {
    handleChange: jest.fn(),
  };
  const wrapper = shallow(<Search handleChange={props.handleChange} />);

  wrapper.find("input").simulate("change");
  expect(props.handleChange).toHaveBeenCalledTimes(1);

  expect(props.handleChange).toHaveBeenCalled();
  expect(wrapper.length).toBe(1);
  expect(wrapper).toMatchSnapshot();
});
