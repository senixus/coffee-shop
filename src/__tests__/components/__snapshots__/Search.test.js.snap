// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Search component 1`] = `
<div
  className="search"
>
  <div>
    <input
      className="search-bar"
      name="coffee"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      placeholder="Search"
      type="text"
    />
  </div>
</div>
`;
